python/cudf/cudf/core/index.py

```diff
@@ -428,8 +433,63 @@ def __array_function__(self, func, types, args, kwargs):
         else:
             return NotImplemented
 
-    def isin(self, values):
-        return self.to_series().isin(values)
+    def isin(self, values, level=None):
+        """Return a boolean array where the index values are in values.
+
+        Compute boolean array of whether each index value is found in
+        the passed set of values. The length of the returned boolean
+        array matches the length of the index.
+
+        Parameters
+        ----------
+        values : set, list-like, Index or Multi-Index
+            Sought values.
+        level : str or int, optional
+            Name or position of the index level to use (if the index
+            is a MultiIndex).
+        Returns
+        -------
+        is_contained : cupy array
+            CuPy array of boolean values.
+
+        """
+
+        if isinstance(values, Index):
```

Comment:
If `values` is a MultiIndex here this will return True and try to convert it to a column erroneously